go/src/socialapi/Makefile

```diff
@@ -54,92 +55,67 @@ build:
+	@env COMPILE_FLAGS="-race" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/workers/algoliaconnector/algoliaconnector
 
 # testcollaboration runs the unit and integration tests of collaboration worker
 testcollaboration:
 	@echo "$(OK_COLOR)--> collaboration tests... $(NO_COLOR)"
-	@`which go` test -c workers/collaboration/tests/*.go
-	@./collaboration.test $(DBG) $(EXTRAS) -c $(CONFIG)
-	@rm  ./collaboration.test
+	@env COMPILE_FLAGS="-race" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/workers/collaboration/tests
 
-	@`which go` test -c workers/collaboration/*.go
-	@./collaboration.test -kite-init $(DBG) $(EXTRAS) -c $(CONFIG)
-	@rm  ./collaboration.test
+	env RUN_FLAGS="-kite-init -c=./go/src/socialapi/config/dev.toml" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/workers/collaboration
 
 testmodels:
 	@echo "$(OK_COLOR)--> model tests... $(NO_COLOR)"
-	@`which go` test -c models/*.go
-	@./models.test $(DBG) $(EXTRAS) -c $(CONFIG)
-	@rm ./models.test
+	@env COMPILE_FLAGS="-race" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/models
 
 testmailsender:
 	@echo "$(OK_COLOR)--> Mailsender tests... $(NO_COLOR)"
-	@`which go` test -c workers/email/emailsender/*.go
-	@./emailsender.test $(DBG) $(EXTRAS) -c $(CONFIG)
-	@rm  ./emailsender.test
+	@env COMPILE_FLAGS="-race" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/workers/email/emailsender
 
 testmail:
 	@echo "$(OK_COLOR)--> mail tests... $(NO_COLOR)"
-	@`which go` test -c workers/email/mailparse/models/*.go
-	@./models.test $(DBG) $(EXTRAS) -c $(CONFIG)
-	@rm ./models.test
+	@env COMPILE_FLAGS="-race" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/workers/email/mailparse/models
 
 	@echo "$(OK_COLOR)--> mail integration tests... $(NO_COLOR)"
-	@`which go` test -c workers/email/mailparse/tests/*.go
-	@./tests.test $(DBG) $(EXTRAS) -c $(CONFIG)
-	@rm ./tests.test
+	@env COMPILE_FLAGS="-race" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/workers/email/mailparse/tests
 
+# testpayments includes -race & -v flags
 testpayment:
 	@echo "$(OK_COLOR)--> payment tests... $(NO_COLOR)"
 	./workers/payment/tests.sh
 
 testintegration:
 	@echo "$(OK_COLOR)--> api integration tests... $(NO_COLOR)"
-	@`which go` test -c tests/*.go
-	@./main.test $(DBG) $(EXTRAS) -c $(CONFIG)
-	@rm ./main.test
+	@env COMPILE_FLAGS="-race" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/tests
 
 testrealtime:
 	@echo "$(OK_COLOR)--> realtime tests... $(NO_COLOR)"
-	@`which go` test -c workers/realtime/realtime/*.go
-	@./realtime.test $(DBG) $(EXTRAS) -c $(CONFIG)
-	@rm ./realtime.test
+	@env COMPILE_FLAGS="-race" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/workers/realtime/realtime
 
 	@echo "$(OK_COLOR)--> realtime model tests... $(NO_COLOR)"
-	@`which go` test -c workers/realtime/models/*.go
-	@./models.test $(DBG) $(EXTRAS) -c $(CONFIG)
-	@rm ./models.test
+	@env COMPILE_FLAGS="-race" $(KODINGDIR)/scripts/gotests.sh socialapi socialapi/workers/realtime/models
```

Comment:
i've added as a note for this '-race' usage in PR description. its just to test step by step for now